devito/symbolics/printer.py

```diff
@@ -31,6 +32,11 @@ def _print_Function(self, expr):
         # There exist no unknown Functions
         if expr.func.__name__ not in self.known_functions:
             self.known_functions[expr.func.__name__] = expr.func.__name__
+        if isinstance(expr, devito.types.Array):
```

Comment:
maybe this should go under a ``_print_Array`` ?

my feeling is that an ``isinstance`` within a visitor handler is a bit odd

Reply:
A separate `_print_Array` doesn't seem to work